src/common/obj_bencher.cc

```diff
@@ -253,13 +253,17 @@ int ObjBencher::aio_bench(
   const std::string run_name_meta = (run_name.empty() ? BENCH_LASTRUN_METADATA : run_name);
 
   //get data from previous write run, if available
-  if (operation != OP_WRITE) {
+  if (operation != OP_WRITE || reuse_bench) {
     uint64_t prev_op_size, prev_object_size;
     r = fetch_bench_metadata(run_name_meta, &prev_op_size, &prev_object_size,
 			     &num_objects, &prevPid);
     if (r < 0) {
-      if (r == -ENOENT)
-        cerr << "Must write data before running a read benchmark!" << std::endl;
+      if (r == -ENOENT) {
+        if (!reuse_bench)
```

Comment:
if directly using 'rados bench 3 write -t 3 -p rbd --reuse-bench', it will report 'Must write data before running a read benchmark', that is confusing because it is a write benchmark

Reply:
i see. could you transpose the error messages? so they look like

```c++
if (reuse_bench) {
  cerr << "Must write data before using reuse_bench for a write benchmark!" << std::endl;
} else {
  cerr << "Must write data before running a read benchmark!" << std::endl;
}
```